### ISSUE TITLE
angular directive selector should allow HtmlTag/custom-component-selector at beginning

### ISSUE THREAD

**Issue by highestop** [angular-eslint/angular-eslint#189]
I think it makes sense to have directive selector like this

```ts
@Directive({
  selector: 'i[appCustomIcon], span[appCustomIcon]',
})
export class CustomIconDirective {}
```

but now with eslint, it encounters an error under this rule

```json
"@angular-eslint/directive-selector": [
    "error",
    { "type": "attribute", "prefix": "app", "style": "camelCase" }
],
```

What I expect is, it will accept a HtmlTag or a custom Component selector at the beginning of one piece of Directive selectors, like `input[appTest]`, but still check the prefix of its own attribute, like `input[appTest]` not `input[test]`

**Comment by JamesHenry** [comment]
How does this work today with TSLint/Codelyzer?

**Comment by Nness** [comment]
This was allowed in TSLint/Codelyzer. It works when without HTML Tag or custom Tag. When add Tag restriction to selector, it throws ESLint error.

Take look at [ng_form](https://github.com/angular/angular/blob/master/packages/forms/src/directives/ng_form.ts#L94) selector and apply angular ESLint to it. 

Working
```
[appCustomIcon]
```

Not Working
```
span[appCustomIcon]
```

The problem is we try to use HTML Tag or custom tag to restrict usage of this directive. The actual selector is attribute, not HTML tag. HTML tag's purpose is serve as restriction.

**Comment by Nness** [comment]
[Codelyzer Test](https://github.com/mgechev/codelyzer/blob/master/test/directiveSelectorRule.spec.ts#L269-L278)

[Codelyzer Rule](https://github.com/mgechev/codelyzer/blob/master/docs/dist/src/directiveSelectorRule.js)

**Comment by GavinWu1991** [comment]
I feel this issue was mentioned in another #166

**Comment by JamesHenry** [comment]
This should hopefully be resolve in `0.8.0-beta.4` thanks to @GavinWu1991
